overlord/snapstate/snapstate.go

```diff
@@ -245,6 +245,63 @@ func Update(s *state.State, name, channel string, userID int, flags Flags) (*sta
 	return doInstall(s, &snapst, ss)
 }
 
+// Enable sets a snap to the active state
+func Enable(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	if snapst.Active {
+		return nil, fmt.Errorf("snap %q already enabled", name)
+	}
+
+	if err := checkChangeConflict(s, name); err != nil {
+		return nil, err
+	}
+
+	ss := &SnapSetup{
+		Name:     name,
+		Revision: snapst.Current,
+	}
+
+	revisionStr := fmt.Sprintf(" (%d)", snapst.Current)
+	prepareSnap := s.NewTask("prepare-snap", fmt.Sprintf(i18n.G("Prepare snap %q%s"), ss.Name, revisionStr))
```

Comment:
The parenthesis is actually quite nice, but I think we're not doing that elsewhere in other tasks. If so, I suggest sticking to:

`"Prepare snap %q revision %s"`
`"Make snap %q revision %s available to the system"`

But maybe we should do a global change towards `"... snap %q (%s) ..."`. Seems much nicer at the task level, which is already more verbose and implies a bit more knowledge of the system.

Note the revision must be a string.